SimpleExec/ProcessExtensions.cs

```diff
@@ -12,13 +14,88 @@ public static void Run(this Process process, bool noEcho, string echoPrefix)
             process.WaitForExit();
         }
 
-        public static Task RunAsync(this Process process, bool noEcho, string echoPrefix)
+        public static async Task RunAsync(this Process process, bool noEcho, string echoPrefix, CancellationToken cancellationToken = default)
         {
             var tcs = new TaskCompletionSource<object>();
-            process.Exited += (s, e) => tcs.SetResult(default);
-            process.EnableRaisingEvents = true;
-            process.EchoAndStart(noEcho, echoPrefix);
-            return tcs.Task;
+
+            using (var started = new SemaphoreSlim(0, 1))
+            using (var ranToCompletionOrCanceled = new SemaphoreSlim(1, 1))
+            using (var cancellation = cancellationToken.Register(() =>
+            {
+                started.Wait();

```

Comment:
`SemaphoreSlim` has `WaitAsync()` if I remember correctly.

Reply:
Yes, but we can't use it here, because `CancellationToken.Register()` only accepts an `Action`.